urlcaef331f4e_dreamable_cra_tgz-pJ8-coho_lp_project_LPProjectJ8/src/coho/interp/SymbolName.java

```diff
@@ -0,0 +1,93 @@
+package coho.interp;
+
+/**
+ * Map the symbol name with symbol key based on a class. The default class is Sym 
+ * generated by CUP
+ */
+// WARNING: the generation of tokenMap and nontermMap is done
+//   by examining the sym class.  I assume that any name that starts
+//   with a capital letter or is the name "error" is the name of a
+//   token, and all other names are non-terminals.  If you introduce
+//   tokens or non-terminals that don't follow these conventions,
+//   toString will print garbage and may generate an Exception.
+
+import java.lang.reflect.Field;
+import java.util.NoSuchElementException;
+import java.util.TreeMap;
+
+public class SymbolName {
+  private static String[] termMap;
+  private static String[] nontermMap;
+  private TreeMap stringToKey;
+
+
+  public SymbolName(Class symClass) {
+    termMap = getMap(symClass, true);
+    nontermMap = getMap(symClass, false);
+    Field[] f = symClass.getDeclaredFields();
+    stringToKey = new TreeMap();
+    try {
+      for(int i = 0; i < f.length; i++)
+        stringToKey.put(f[i].getName(), new Integer(f[i].getInt(symClass)));
+    } catch (IllegalAccessException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)